### ISSUE TITLE
Local test network startup sometimes fails, proxy unable to find nodes

### ISSUE THREAD

**Issue by hiddentao** [ElrondNetwork/elrond-sdk#112]
Sometimes, when I try to run a local test network the proxy is unable to find the nodes and the log looks like:

```
INFO:scope:running in local testnet
INFO:testnet:using local testnet config from /home/erd/sandbox/testnet.toml
INFO:testnet:sdk_testnet_config filename /home/erd/elrondsdk/testnet.toml
INFO:scope:initialized. chain = local-testnet, proxy = http://localhost:7950
INFO:cli.testnet:Starting testnet...
INFO:testnet:using local testnet config from testnet.toml
INFO:testnet:sdk_testnet_config filename /home/erd/elrondsdk/testnet.toml
INFO:testnet:testnet folder is /home/erd/sandbox/testnet
Started process [151] ['./seednode', '--log-save']
[PID=151] [GIN-debug] [WARNING] Creating an Engine instance with the Logger and Recovery middleware already attached.
[PID=151] [GIN-debug] [WARNING] Running in "debug" mode. Switch to "release" mode in production.
[PID=151] WARN [2020-12-15 14:57:06.745]   nil or empty initial peers list provided to kad dht implementation. No initial connection will be done
Started process [162] ['./node', '--use-log-view', '--log-save', '--log-level=*:DEBUG,arwen:TRACE,process/smartcontract:TRACE', '--log-logger-name', '--log-correlation', '--rest-api-interface=localhost:10100']
Started process [164] ['./node', '--use-log-view', '--log-save', '--log-level=*:DEBUG,arwen:TRACE,process/smartcontract:TRACE', '--log-logger-name', '--log-correlation', '--rest-api-interface=localhost:10101']
[PID=162] WARN [2020-12-15 14:57:08.783] [update/trigger]     [/0/0/]        version string is not in the correct format. Hardfork trigger will lead to unpredictable results
[PID=164] WARN [2020-12-15 14:57:08.796] [update/trigger]     [/0/0/]        version string is not in the correct format. Hardfork trigger will lead to unpredictable results
Started process [185] ['./proxy']
[PID=185] [GIN-debug] [WARNING] Creating an Engine instance with the Logger and Recovery middleware already attached.
[PID=185] [GIN-debug] [WARNING] Running in "debug" mode. Switch to "release" mode in production.
[PID=185] ERROR[2020-12-15 14:57:16.687]   validator statistics                     observer = http://localhost:10101 error = no response
[PID=185] ERROR[2020-12-15 14:57:16.687]   heartbeat                                observer = http://localhost:10101 error = no response
[PID=185] WARN [2020-12-15 14:57:16.687]   validator statistics: get from API       error = validator statistics data not found at any observer
[PID=185] ERROR[2020-12-15 14:57:16.687]   heartbeat                                observer = http://localhost:10100 error = no response
[PID=185] WARN [2020-12-15 14:57:16.687]   heartbeat: get from API                  error = heartbeat status not found at any observer
[PID=185] ERROR[2020-12-15 14:57:41.655]   heartbeat                                observer = http://localhost:10100 error = no response
[PID=185] ERROR[2020-12-15 14:57:41.657]   heartbeat                                observer = http://localhost:10101 error = no response
[PID=185] WARN [2020-12-15 14:57:41.657]   heartbeat: get from API                  error = heartbeat status not found at any observer
[PID=185] ERROR[2020-12-15 14:58:06.624]   heartbeat                                observer = http://localhost:10101 error = no response
[PID=185] ERROR[2020-12-15 14:58:06.625]   heartbeat                                observer = http://localhost:10100 error = no response
[PID=185] WARN [2020-12-15 14:58:06.625]   heartbeat: get from API                  error = heartbeat status not found at any observer
[PID=164] ERROR[2020-12-15 14:58:08.847] [main]               [/0/0/]        bootstrap return error                   error = timeout while waiting for epoch start meta block
[PID=164] ERROR[2020-12-15 14:58:08.847] [main]               [/0/0/]        timeout while waiting for epoch start meta block
[PID=162] ERROR[2020-12-15 14:58:08.849] [main]               [/0/0/]        bootstrap return error                   error = timeout while waiting for epoch start meta block
[PID=162] ERROR[2020-12-15 14:58:08.849] [main]               [/0/0/]        timeout while waiting for epoch start meta block
Proces [164] stopped. Return code: 1.
Proces [162] stopped. Return code: 1.
[PID=185] ERROR[2020-12-15 14:58:16.620]   validator statistics                     observer = http://localhost:10101 error = no response
[PID=185] WARN [2020-12-15 14:58:16.620]   validator statistics: get from API       error = validator statistics data not found at any observer
...
```

This happens I can confirm that the `node` processes are initially running, though it appears from the logs that they both exit badly early on. 

I am trying to run the test network inside a [Docker container](https://github.com/erdDEVcode/narya/blob/master/Dockerfile). Host machine is OS X 10.15.6, 2.3 GHz Quad-Core Intel Core i7, 32 GB RAM.

**Comment by hiddentao** [comment]
Seeing this in node logs for `validator00`:

```
DEBUG[2020-12-15 15:44:26.795] [..hStart/bootstrap] [/0/0/]        no epoch db found in storage             error = open /home/erd/sandbox/testnet/validator00/db/local-testnet: no such file or directory 

...

DEBUG[2020-12-15 15:44:26.839] [../requesthandlers] [/0/0/]        requesting header by epoch               topic = metachainBlocks epoch = 4294967295 hash = 65706f63685374617274426c6f636b5f34323934393637323935 
DEBUG[2020-12-15 15:44:26.839] [../requesthandlers] [/0/0/]        RequestStartOfEpochMetaBlock.RequestDataFromEpoch error = cannot send request: peer list is empty or errors during the sending, topic: metachainBlocks, crossPeers: 0, intraPeers: 0 epochStartIdentifier = 65706f63685374617274426c6f636b5f34323934393637323935 

...

DEBUG[2020-12-15 15:45:25.709] [p2p/libp2p]         [/0/0/]        network connection metrics               connections/s = 0 disconnections/s = 0
DEBUG[2020-12-15 15:45:25.823] [../requesthandlers] [/0/0/]        requesting header by epoch               topic = metachainBlocks epoch = 4294967295 hash = 65706f63685374617274426c6f636b5f34323934393637323935
DEBUG[2020-12-15 15:45:26.770] [..hStart/bootstrap] [/0/0/]        unregistering all message processor and un-joining all topics
ERROR[2020-12-15 15:45:26.771] [main]               [/0/0/]        bootstrap return error                   error = timeout while waiting for epoch start meta block
DEBUG[2020-12-15 15:45:26.771] [p2p/libp2p]         [/0/0/]        closed subscription                      topic = metachainBlocks err = subscription cancelled
ERROR[2020-12-15 15:45:26.771] [main]               [/0/0/]        timeout while waiting for epoch start meta block

```

For `validator01` I see:

```
DEBUG[2020-12-15 15:44:26.795] [..hStart/bootstrap] [/0/0/]        no epoch db found in storage             error = open /home/erd/sandbox/testnet/validator01/db/local-testnet: no such file or directory 

...

DEBUG[2020-12-15 15:45:25.820] [../requesthandlers] [/0/0/]        requesting header by epoch               topic = metachainBlocks epoch = 4294967295 hash = 65706f63685374617274426c6f636b5f34323934393637323935 
DEBUG[2020-12-15 15:45:26.774] [..hStart/bootstrap] [/0/0/]        unregistering all message processor and un-joining all topics 
ERROR[2020-12-15 15:45:26.774] [main]               [/0/0/]        bootstrap return error                   error = timeout while waiting for epoch start meta block 
ERROR[2020-12-15 15:45:26.774] [main]               [/0/0/]        timeout while waiting for epoch start meta block 
```

**Comment by hiddentao** [comment]
Can confirm after much use that I have to rebuild my docker image from scratch on every run. That's the only way it works. And on the rare occasion that I've been able to run a container from the same image a second time, I've noticed that the epoch/round number continues onwards from the previous run. 

This is obviously weird since each Docker container is supposed to start with a fresh virtual disk based on the original image, with no link to any previously run containers.

I suspect that the local testnet is registering itself with some server somewhere with some sort of unique id? And that this is why subsequent runs fail. Rebuilding the image from scratch resets the unique id. This is my theory anyway.
